On Tue, Oct 21, 2003 at 03:13:38PM -0700, Greg KH wrote:
> On Tue, Oct 21, 2003 at 02:45:55PM -0700, Jesse Barnes wrote:
> > Thanks for the new release, Greg.  I just tried it out on a system with
> > some disks, but a bunch of udev processes ended up hanging.  Is there
> > something I'm missing or do you need a patch like this?
> 
> Yeah, sorry, this kind of fix is required :(
> 
> It's fixed in my bk tree now.
> 
> Oh, and it looks like the LABEL rule is also broken due to the libsysfs
> changes...  I'm working on adding regression tests right now to prevent
> things like this from slipping through.

Here's a patch for this.  It fixes the problem of LABEL rules on the
device, not the class device.  LABEL rules on the class device seem to
work just fine.

Thanks to Dan Stekloff for help in finding this bug.

thanks,

greg k-h


# fix LABEL bug for device files (not class files.)

diff -Nru a/namedev.c b/namedev.c
--- a/namedev.c	Tue Oct 21 18:44:59 2003
+++ b/namedev.c	Tue Oct 21 18:44:59 2003
@@ -566,7 +566,7 @@
 
 			/* look in the class device directory if present */
 			if (class_dev->sysdevice) {
-				tmpattr = sysfs_get_classdev_attr(class_dev, dev->sysfs_file);
+				tmpattr = sysfs_get_device_attr(class_dev->sysdevice, dev->sysfs_file);
 				if (tmpattr)
 					goto label_found;
 			}
@@ -599,7 +599,7 @@
 
 					/* look in the class device directory if present */
 					if (class_dev_parent->sysdevice) {
-						tmpattr = sysfs_get_classdev_attr(class_dev_parent, dev->sysfs_file);
+						tmpattr = sysfs_get_device_attr(class_dev_parent->sysdevice, dev->sysfs_file);
 						if (tmpattr) 
 							goto label_found;
 					}
-
To unsubscribe from this list: send the line "unsubscribe linux-kernel" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html
Please read the FAQ at  http://www.tux.org/lkml/